src/server.h

```diff
@@ -887,20 +890,28 @@ typedef struct {
     uint64_t flags; /* See USER_FLAG_* */
 
     /* The bit in allowed_commands is set if this user has the right to
-     * execute this command. In commands having subcommands, if this bit is
-     * set, then all the subcommands are also available.
+     * execute this command.
      *
      * If the bit for a given command is NOT set and the command has
-     * subcommands, Redis will also check allowed_subcommands in order to
+     * allowed first-args, Redis will also check allowed_firstargs in order to
      * understand if the command can be executed. */
     uint64_t allowed_commands[USER_COMMAND_BITS_COUNT/64];
 
-    /* This array points, for each command ID (corresponding to the command
+    /* NOTE: allowed_firstargs is a relic of the old mechanism for allowing
+     * subcommands (now, subcommands are actually commands, with their own
+     * ACL ID)
+     * We had to keep allowed_firstargs (previously called allowed_subcommands)
+     * in order to support the widespread abuse of ACL rules to block a command
+     * with a specific argv[1] (which is not a subcommand at all).
+     * For example, a user can use the rule "-select +select|0" to block all
+     * SELECT commands, except "SELECT 0".
```

Comment:
i'm not sure we wanna mention it's a relic (now that we gave it a proper name).
we can however say that it can be applied on a first arg of a sub-command, i.e. it could be argv[2]

Reply:
I didn't have anything specific against the word "relic". i meant that i don't think we should document the history and how it was named, and that "We had to keep allowed_firstargs (previously called allowed_subcommands)".
just document what it does and that it can be either `argv[1]` or `argv[2]` (when applied on a sumbommand)